addons/sale_project/models/project.py

```diff
@@ -394,20 +409,83 @@ def _get_revenues_items_from_sol(self, domain=None, with_action=True):
+
+    def _get_items_from_aal(self, with_action=True):
+        def get_action(invoice_type, record_ids):
+            args = [invoice_type, [('id', 'in', record_ids)]]
+            if len(record_ids) == 1:
+                args.append(record_ids[0])
+            return {'name': 'action_aal', 'type': 'object', 'args': json.dumps(args)}
+
+        domain = self._get_aal_domain()
+        aal_other_search = self.env['account.analytic.line'].search(domain)
+        if not aal_other_search:
+            return None
+        # The account.groupe_account_readonly is set in the account_accountant module. There's no bridge between this module and the sale_project.
+        # Therefore we add this security check-up here because the use case can not be made without the accounting module.
+        can_see_aal = with_action and len(self) == 1 and self.user_has_groups('account.group_account_readonly')
+        total_revenues = total_costs = 0.0
+        costs = revenues = self.env['account.analytic.line']
+        for aal in aal_other_search:
+            aal_amount = aal.amount
+            if aal_amount < 0.0:
+                total_costs += aal_amount
+                costs |= aal
+            else:
+                total_revenues += aal_amount
+                revenues |= aal
+        costs_action = None
+        revenues_action = None
+        if can_see_aal:
+            costs_action = get_action('other_costs', costs.ids)
+            revenues_action = get_action('other_revenues', revenues.ids)
+        # we dont know what part of the numbers has already been billed or not, so we have no choice but to put everything under the billed/invoiced columns.
+        # The to bill/to invoice ones will simply remain equal to 0, as it's already the case for other items in the right-side panel
+        return {
+            'revenues': {'id': 'other_revenues', 'sequence': 12, 'invoiced': total_revenues, 'to_invoice': 0.0, 'action': revenues_action},
+            'costs': {'id': 'other_costs', 'sequence': 13, 'billed': total_costs, 'to_bill': 0.0, 'action': costs_action},
+        }
+
     def _get_profitability_items(self, with_action=True):
         profitability_items = super()._get_profitability_items(with_action)
         revenue_items_from_sol = self._get_revenues_items_from_sol(
             [('order_id', 'in', self.sudo()._get_sale_orders().ids)],
             with_action,
         )
         revenues = profitability_items['revenues']
+        # The items from aal are only computed if there is only one project linked to the so, otherwise, we'd have to show the same extra cost/revenues on all project linked.
+        # which is inaccurate.
+        items_from_aal = None
+        if self.allow_billable and len(self.analytic_account_id.project_ids) == 1:
+            items_from_aal = self._get_items_from_aal(with_action)
+        if items_from_aal:
+            costs = profitability_items['costs']
```

Comment:
In the end, we only have `items_from_aal` if the first condition.
```suggestion
        if self.allow_billable and len(self.analytic_account_id.project_ids) == 1:
            items_from_aal = self._get_items_from_aal(with_action)
            costs = profitability_items['costs']
```

Reply:
No, there's the possibility for _get_items_from_all to return None, if no aal were found. In that case, we dont want to show the other_cost/revenues line in the panel. But i can indeed put the "if items_from_aal" in the first if so we dont check the second if if the first was not fulfilled.

Reply:
My bad, you're right!